### DISCUSSION TITLE
Auto Trim Image does not work on some images in version 4.9.3

### DISCUSSION THREAD

**Discussion by Tribo-2dot0** [arkypita/LaserGRBL#1895] [General]
You click Auto Trim and nothing happens. 
I converted to png and it's still the same.
One of the images in question is this one:

![tigre vetor](https://user-images.githubusercontent.com/83982525/184499786-755754af-a6f5-47be-8888-4fcbc5bb8444.jpg)

In another image, which Auto Trim didn't work either, I tried to use Crop Image but it gave an error (which I forgot to copy and now I don't remember which image), the one in the Ooooops window...
If the error appears again, copy and paste here.

**Comment by StuartB4** [comment]
Works fine for me with 4.9.3. Tried all the different modes and it did it on all of them.

**Reply by Tribo-2dot0** [reply]
For me all worked, except this image and another one that has an error Oooops...
Did you try this tiger image?

**Comment by StuartB4** [comment]
I tried the tiger as a jpg and it did'nt work, try converting it to a .png file, it works for me in all modes.

**Comment by StuartB4** [comment]
If that does'nt work try removing the background with a website called https://www.remove.bg
It will remove a lot of the white around the tiger and save it as a .png file.

**Comment by StuartB4** [comment]
![image](https://user-images.githubusercontent.com/16980026/184635222-f803eb89-4d66-4691-91cc-058f6cf7a32e.png)

**Reply by Tribo-2dot0** [reply]
I just took this image of the tiger to show how the new "Outline Image" and "Flood Fill" functions work in a video, but then this happened and as an error had already occurred, in another image, when using "Crop Image", I resolved make this post in order to report a possible bug in the new version.

**Comment by arkypita** [comment]
Your image have borders. It is normal that the automatic trim does not trim it.

![image](https://user-images.githubusercontent.com/8782035/184714939-2a3e7625-a47d-4c11-8fa9-70abf2c72559.png)

![image](https://user-images.githubusercontent.com/8782035/184715254-d092e607-27ce-4a1a-bbc3-eef9956f3650.png)

**Reply by Tribo-2dot0** [reply]
Oh ok... I hadn't noticed. Sorry for my mistake.
It's just that I was focused on another error that occurred using Crop Image, in another image, and I thought it was related.
When I try to Crop this image:

![Captura de Tela (138)](https://user-images.githubusercontent.com/83982525/184718791-32c73628-9edb-40d5-8053-506630d64824.png)

This error appears:

![Captura de Tela (137)](https://user-images.githubusercontent.com/83982525/184718848-ac3c5e01-2dce-461c-9bcd-01b32c7eaed0.png)

**Reply by arkypita** [reply]
please share the image (zip it, so there is no compression by github) so I can test it

**Reply by Tribo-2dot0** [reply]
[logo tribo.zip](https://github.com/arkypita/LaserGRBL/files/9340318/logo.tribo.zip)

**Reply by arkypita** [reply]
I tested with your image file and I had no problems, tested twenty times with different crops (small, large, left to right, top to bottom etc) no exceptions and no unexpected memory use (memory usage about 30MB).

**Comment by arkypita** [comment]
Why in this image there is a red crop frame?
https://user-images.githubusercontent.com/83982525/184718791-32c73628-9edb-40d5-8053-506630d64824.png

**Reply by Tribo-2dot0** [reply]
I did it with MS Paint to mark the area where I tried to make the Crop... I started from the bottom left to the top right.

**Reply by arkypita** [reply]
Ok, found. I was able to repeat. You have this exception when you try to crop outside the image limits, because the rectangle is not correctly limited and allow to select outside the image. I'll fix in next version of LaserGRBL.

![image](https://user-images.githubusercontent.com/8782035/184909290-dc0d262e-8258-4290-89ec-7bd0496ed3c0.png)

**Reply by arkypita** [reply]
Moved here: https://github.com/arkypita/LaserGRBL/issues/1902
